### ISSUE TITLE
[Show and Tell] Jaxonloader - A dataloader for JAX built as eqx.Module

### ISSUE THREAD

**Issue by Artur-Galstyan** [patrick-kidger/equinox#672]
Hi there, since the label `show and tell` now exists, I felt encouraged to share a little something I've been cooking for a couple of weeks. 

It's called [Jaxonloader](https://github.com/Artur-Galstyan/jaxonloader) and is a dataloader for JAX. Under the hood, it's a stateful `eqx.Module`, which returns the current row of the data (basically). It's quite a thin wrapper around JAX arrays.

The goal is to add more and more public datasets and to not rely on PyTorch's dataloader anymore. The conversion from `torch.Tensor` to `jax.Array` always pained me and I don't want to have a PyTorch dependency in my JAX project.

Here's a quickstart:

```python
import jax
from jaxonloader import get_mnist, make

key = jax.random.PRNGKey(0)

train, test = get_mnist()
# these are JaxonDatasets

train_loader, index = make(
    train,
    batch_size=4,
    shuffle=True,
    drop_last=True,
    key=key,
    jit=True
)
while x:= train_loader(index):
    data, index, done = x
    processed_data = process_data(data)
    if done:
        break
```

Cheers!
